public/pages/AnomalyCharts/containers/AnomalyDetailsChart.tsx

```diff
@@ -351,17 +351,29 @@ export const AnomalyDetailsChart = React.memo(
 
     const customAnomalyContributionTooltip = (details?: string) => {
       const anomaly = details ? JSON.parse(details) : undefined;
-      const contributionData = get(anomaly, `features`, {})      
-      const featureAttributionList = [];
-      for (const [key, value] of Object.entries(contributionData)) {
-       const attribution = Number(JSON.stringify(value.data))
-       featureAttributionList.push(
-          <div>
-            {key}: {attribution} <br />
-          </div>
-        )
-      }
+      const contributionData = get(anomaly, `contributions`, [])  
 
+      const featureData = get(anomaly, `features`, {}) 
+      let featureAttributionList = [] as any[];
+      if (Array.isArray(contributionData)) {
```

Comment:
when would contributions not be an array?

Reply:
when it's getting response by calling search result api route in stead of get detector result api route

Reply:
Not sure what you mean about get detector result api route. I see the parsing you've done in `getAnomalyResults()`. Where does the parsing happen where it is an array instead?

Reply:
there's no parsing for that route because it's processed at receiving response stage.  See line [964](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/296/files#diff-cc594889ac44cb130233969adf3da57a18f360a2a77c57ee273aad5f39d6d733R964) in ad.ts file, a type was defined and returned with response data

Reply:
I'm still unsure when contributions would be an array. Even `AnomalyResult` defines it as a map [here](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/296/files#diff-71f2b9246d36bd7c633e08d58e01d11cc08532db25bb0e00fd1efe2aad804751R116)

Reply:
it's from here - https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/main/public/pages/DetectorResults/containers/AnomalyHistory.tsx#L664

Reply:
the data has been flattened before passing in - [https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/296/files#diff-45683a46376000d3a5181f2[…]457590f0212f08b208877beaR602](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/296/files#diff-45683a46376000d3a5181f2d2b68401f594dfe84457590f0212f08b208877beaR602)